File: api-cmssignature/api-mssclient/mssclient/src/tr/gov/tubitak/uekae/esya/api/webservice/mssclient/wrapper/EMSSPException.java
package tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper;

import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.api.infra.mobile.Status;

public class EMSSPException extends ESYAException
{
    private Status status;

    public EMSSPException(String aMessage)
    {
        super(aMessage);
    }

    public EMSSPException(Status aResultStatus)
    {
        this.status = aResultStatus;
    }

    public EMSSPException(String aMessage,Status aResultStatus)
    {
        super(aMessage);
        this.status = aResultStatus;
    }

    public EMSSPException(String aMessage,Status aResultStatus,Exception aCause)
    {
        super(aMessage);
        this.status = aResultStatus;
    }

    public EMSSPException(String aMessage, Exception aCause)
    {
        super(aMessage,aCause);

    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public EMSSPException(Status aResultStatus,Exception aCause)
    {
        super(aCause);
        this.status = aResultStatus;
    }

    @Override
    public String toString() {
        return "EMSSPException{" +
                "cause=" + super.toString() +
                ", status=" + status +
                '}';
    }
}

File: api-cmssignature/api-mssclient/mssclient/csharp/src/wrapper/EMSSPException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using tr.gov.tubitak.uekae.esya.api.common;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper;

namespace tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper
{
    public class EMSSPException : ESYAException
    {
        private Status status;


        public EMSSPException(String aMessage)
            : base(aMessage)
        {
        }

        public EMSSPException(Status aResultStatus)
        {
            this.status = aResultStatus;
        }

        public EMSSPException(String aMessage,Status aResultStatus):base(aMessage)
        {
            this.status = aResultStatus;
        }

        public EMSSPException(String aMessage,Status aResultStatus,Exception aCause):base(aMessage, aCause)
        {
            this.status = aResultStatus;
        }

        public EMSSPException(String aMessage, Exception aCause)
            : base(aMessage, aCause)
        {
            
        }
        public Status getStatus()
        {
            return status;
        }

        public void setStatus(Status status)
        {
            this.status = status;
        }
        public EMSSPException(Status aResultStatus,Exception aCause)
            : base(aCause.Message, aCause)
        {
            this.status = aResultStatus;
        }
    }
}
